client/normal_nil.go

```diff
@@ -34,13 +34,13 @@ func NewNormalNil(kind FieldKind) (NormalValue, error) {
 		return NewNormalNillableString(immutable.None[string]()), nil
 	case FieldKind_NILLABLE_BLOB:
 		return NewNormalNillableBytes(immutable.None[[]byte]()), nil
-	case FieldKind_NILLABLE_BOOL_ARRAY:
+	case FieldKind_BOOL_ARRAY, FieldKind_NILLABLE_BOOL_ARRAY:
```

Comment:
todo: This should be changed to 
```go
case FieldKind_BOOL_ARRAY:
    return NewNormalIntNillableArray(immutable.None[[]bool]()), nil
case FieldKind_NILLABLE_BOOL_ARRAY:
	return NewNormalNillableBoolNillableArray(immutable.None[[]immutable.Option[bool]]()), nil
```
and similarly for the other array types.

Reply:
Nice spot :)